include/deal.II/base/patterns.h

```diff
@@ -1162,6 +1162,10 @@ namespace Patterns
         else if (std::is_floating_point<T>::value)
           return std_cxx14::make_unique<Patterns::Double>(
                    -std::numeric_limits<T>::max(), std::numeric_limits<T>::max());
+
+        Assert(false, ExcNotImplemented());
+        //the following line should never be invoked
+        return std_cxx14::make_unique<Patterns::Anything>();
```

Comment:
Hm, I think the better approach would be to return a `nullptr` here instead of a pointer to an `Anything`. What do you think?

Reply:
Sure, `nullptr` is certainly more invalid than `Anything`.

Reply:
The documentation implies that the function always returns a pointer to a valid object, so I would personally prefer to return a pointer to `Patterns::Anything`.

Reply:
Not that it really matters, since its a bug if we ever get that far anyway.

Reply:
I'm thinking that with a `nullptr`, you'd at least get a segfault if you happened to run your program in release mode and did something wrong. That's better than getting wrong results.

Reply:
I agree. It's quite easy to forget to test in `Debug`mode first...

Reply:
It seems to happen to 50% of our users on a regular basis :-)